snappy/firstboot_test.go

```diff
@@ -118,8 +112,17 @@ func (s *FirstBootTestSuite) newPartMap() (map[string]Part, error) {
 	return s.partMap, s.partMapErr
 }
 
-func (s *FirstBootTestSuite) newFakeApp() *fakePart {
-	fakeMyApp := fakePart{snapType: snap.TypeApp}
+func (s *FirstBootTestSuite) newOverlord() configurator {
+	return s.fakeOverlord
+}
+
+func (s *FirstBootTestSuite) newFakeApp() *SnapPart {
+	fakeMyApp := SnapPart{
+		m: &snapYaml{
```

Comment:
Thanks, I agree and this is a good example why it can slow down the development. I hit a similar problem when removing Details() from the repo interface.